### PR TITLE
[Cisco ASA] Support patterns with SGT tag:name as username

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
When using Cisco ASA with the SXP protocol and SGT-tags to write firewall rules, the format of the [`idfw_user`](https://www.cisco.com/c/en/us/td/docs/security/asa/syslog/b_syslog/syslog-messages-302003-to-342008.html#con_6941209) is presented in their logs as `SGT-tag:SGT-name`. Multiple message_id are now required to support this change. 

Sample log: `<140>Oct 06 2023 09:22:19 myAsaHostname : %ASA-4-106023: Deny tcp src outside:192.168.2.2/51982(9999:my_SgtName) dst inside:192.168.2.3/443 by access-group \"outside_access_in\" [0x2a9e189a, 0x0]` which will fail with existing GROK patterns as `(9999:my_SgtName)` is not handled.

Fixed by updating `CISCO_USER` pattern in each of these messages to support `SGT-tag:SGT-name`

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates https://github.com/elastic/sdh-beats/issues/3904

### CODE REVIEW

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -384,12 +385,14 @@ processors:
...
+        - ^%{NOTSPACE:event.outcome} ((protocol %{POSINT:network.iana_number})|%{NOTSPACE:network.transport}) src %{NOTCOLON:_temp_.cisco.source_interface}:%{IPORHOST:source.address}(/%{NUMBER:source.port})?\s*(\(%{CISCO_USER_OR_SGT:_temp_.cisco.source_user_or_sgt}\) )?dst %{NOTCOLON:_temp_.cisco.destination_interface}:%{IPORHOST:destination.address}(/%{NUMBER:destination.port})?%{DATA}by access-group "%{NOTSPACE:_temp_.cisco.list_id}"
       pattern_definitions:
         HOSTNAME: "\\b(?:[0-9A-Za-z][0-9A-Za-z-_]{0,62})(?:\\.(?:[0-9A-Za-z][0-9A-Za-z-_]{0,62}))*(\\.?|\\b)"
         IPORHOST: "(?:%{IP}|%{HOSTNAME})"
         NOTCOLON: "[^:]*"
-        CISCO_USER: (\*\*\*\*\*|((LOCAL\\)?(%{HOSTNAME}\\)?%{USERNAME}(@%{HOSTNAME})?(, *%{NUMBER})?))
+        CISCO_USER_OR_SGT: (%{CISCO_USER}|%{CISCO_SGT})
+        CISCO_USER: (\*\*\*\*\*|((LOCAL\\)?(%{HOSTNAME}\\)?%{USERNAME}(@%{HOSTNAME})?%{CISCO_SGT}?))
+        CISCO_SGT: (, *)?%{NUMBER}(:%{WORD})?
```
Comment (efd6) [inline packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:395]:
```suggestion
        CISCO_USER_OR_SGT: (?:%{CISCO_USER}|%{CISCO_SGT})
        CISCO_USER: (?:\*\*\*\*\*|(?:(?:LOCAL\\)?(?:%{HOSTNAME}\\)?%{USERNAME}(?:@%{HOSTNAME})?%{CISCO_SGT}?))
        CISCO_SGT: (?:, *)?%{NUMBER}(?::%{WORD})?
```
(similar below)

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -1681,19 +1698,37 @@ processors:
...
+            message: "Processor {{ _ingest.on_failure_processor_type }} with tag {{ _ingest.on_failure_processor_tag }} in pipeline {{ _ingest.on_failure_pipeline }} failed with message: {{ _ingest.on_failure_message }}"
+  # Parse Source without User only containing SGT
+  - grok:
+      field: "_temp_.cisco.source_sgt"
+      tag: "grok_cisco_source_sgt"
+      if: 'ctx?._temp_?.cisco?.source_sgt != null'
+      patterns:
+        - '%{CISCO_SGT}'
+      pattern_definitions:
+        CISCO_SGT: (, *)?(%{NUMBER:_temp_.cisco.source_user_security_group_tag})?(:)?%{WORD:_temp_.cisco.source_user_security_group_tag_name}?
```
Comment (efd6) [inline packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:1731]: Why is the `(:)` needed? Can't it be just `:`? (similar below)

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (kcreddy) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: updated as suggested

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (kcreddy) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: Its not needed. Updated as suggested

Comment (efd6) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: Is that colon really optional? I don't see any test cases where there is no colon. Are you using this as a shortcut for `(tag|name|tag:name)`?

**Review by efd6 — APPROVED** [review]

**Pull request merged**